app/src/utils/walletStore.ts

```diff
@@ -1,393 +1,350 @@
-import { WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
 import type {
-	MessageSignerWalletAdapter,
-	MessageSignerWalletAdapterProps,
-	SendTransactionOptions,
-	SignerWalletAdapter,
-	SignerWalletAdapterProps,
-	WalletError
+    Adapter,
+    MessageSignerWalletAdapter,
+    MessageSignerWalletAdapterProps,
+    SendTransactionOptions,
+    SignerWalletAdapter,
+    SignerWalletAdapterProps,
+	WalletReadyState,
+    WalletError,
+    WalletName,
 } from '@solana/wallet-adapter-base';
-import type { Wallet, WalletName } from '@solana/wallet-adapter-wallets';
+import { WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
 import type { Connection, PublicKey, Transaction, TransactionSignature } from '@solana/web3.js';
 import { get, writable } from 'svelte/store';
 import { WalletNotSelectedError } from './errors';
 import { getLocalStorage, setLocalStorage } from './localStorage';
 
-type Adapter = ReturnType<Wallet['adapter']>;
-type WalletDictionary = { [name in WalletName]: Wallet };
+
 type ErrorHandler = (error: WalletError) => void;
+type WalletConfig = Pick<WalletStore, 'wallets' | 'autoConnect' | 'localStorageKey' | 'onError'>;
+type WalletStatus = Pick<WalletStore, 'connected' | 'publicKey'>;
 
-interface WalletConfigStore {
-	wallets: Wallet[];
-	walletsByName: WalletDictionary;
-	autoConnect: boolean;
-	localStorageKey: string;
-	onError: ErrorHandler;
-}
+// export interface Wallet {
+//     adapter: Adapter;
+//     readyState: WalletReadyState;
+// }
 
 interface WalletStore {
-	wallet: Wallet | null;
-	publicKey: PublicKey | null;
-	ready: boolean;
-	connected: boolean;
-	connecting: boolean;
-	disconnecting: boolean;
-
-	select(walletName: WalletName): void;
-
-	connect(): Promise<void>;
-
-	disconnect(): Promise<void>;
-
-	sendTransaction(
-		transaction: Transaction,
-		connection: Connection,
-		options?: SendTransactionOptions
-	): Promise<TransactionSignature>;
-
-	signTransaction: SignerWalletAdapterProps['signTransaction'] | undefined;
-	signAllTransactions: SignerWalletAdapterProps['signAllTransactions'] | undefined;
-	signMessage: MessageSignerWalletAdapterProps['signMessage'] | undefined;
+	// props
+    autoConnect: boolean;
+    wallets: Adapter[];
```

Comment:
This is the biggest change here. Following what the Vue core is doing to be updated in https://github.com/solana-labs/wallet-adapter/pull/267/files, it seems that any intern reference to wallet is removed and Adapter is the main property to look at to see all the changes in the store.